### ISSUE TITLE
consolidate.js

### ISSUE THREAD

**Issue by ForbesLindesay** [karthikv/nodefront#6]
Could you use [consolidate.js](https://github.com/visionmedia/consolidate.js) to support a much wider variety of templating engines?

**Comment by karthikv** [comment]
Ah, I've never heard of consolidate.js until now, but it looks promising. I'll see what I can do.

**Comment by karthikv** [comment]
By the way, thanks for all these issues! It's my pleasure to fix them and improve nodefront.

**Comment by ForbesLindesay** [comment]
This was similar to something I started looking at producing myself, and you're further ahead than I was, so I figure it makes more sense to improve this.  consolidate.js is the recommended way of doing view engines in express now, so it's pretty popular.  You could support the same system as we do for marked (and document it) where people who want to be able to use say `ejs` templates in their site can run `npm install ejs -g`.

I'm thinking of possibly creating a module that would be like consolidate, but include support for languages that aren't template languages, such as less, stylus, markdown,  coffee-script.  My plan would be to exactly match the consolidate.js api (and include all consolidate.js's rendering functions as well)  Would you be interested in including that in this app if I built it?

**Comment by karthikv** [comment]
Yes, definitely! As long as the languages fall under front-end development, that'll be fine. I don't want to extend the scope of nodefront to the backend as I feel it'll take away from the focus of the project.

**Comment by ForbesLindesay** [comment]
Yeh, my thinking for nodefront would be to ultimately use it as a static site generator for github.

**Comment by karthikv** [comment]
Ah, yes, I've heard a similar idea from others and I'm quite interested in the possibilities. It's certainly a nice use case for nodefront.

**Comment by ForbesLindesay** [comment]
There are some more features that may be useful in the future with that in mind:
- plugin arcitecture of some sort
- (yaml based?) metadata
- something to allow markdown to have a layout template in another language.

**Comment by ForbesLindesay** [comment]
I've now created [consolidate-build](https://github.com/ForbesLindesay/consolidate-build) which is essentially a much extended version of consolidate.js  It includes the original consolidate, so I recommend we use that instead for nodefront if possible.

**Comment by karthikv** [comment]
Looks good! I plan on adding this in tomorrow and upping nodefront to version 0.1.0 afterwards. As noted in your first commit message, you should definitely add tests in for consolidate-build ASAP.

**Comment by ForbesLindesay** [comment]
Tests added.  Coffee-Script doesn't work yet.  But I'll see if I can get it fixed tomorrow.

**Comment by karthikv** [comment]
Great! I'll be sure to add you under the contributors section after I integrate consolidate-build.

**Comment by karthikv** [comment]
I've just pushed out a version of nodefront with consolidate-build and I'm liking how the code is much cleaner for compilation (see commit above). Waiting on CoffeeScript support before I release version 0.1.0, as some tests are failing due to it. Let me know when you have that integrated.

Thanks for the contribution!

**Comment by ForbesLindesay** [comment]
I've fixed coffee script support and added in a unit test for it.  I've force published it as 0.0.0.  From now on I'll go with:
- The major version will increase for any backward-incompatible changes.
- The minor version will increase for added languages.
- The patch version will increase for bug-fixes.

I've added you as a contributor so that if this becomes a more popular library you can also rapidly contribute any languages you need to or accept pull requests.  Please only add languages that are NOT templates as templates can be added to consolidate.js instead.

**Comment by karthikv** [comment]
Well done.

I released version 0.1.0 with consolidate-build integrated and added you under the contributors section. The next issue to tackle seems to be the metadata or plugin architecture you mentioned earlier. It'd be better to create a new issue for that, so I'll close this one shortly.

Would you like to discuss some of the ideas you have for the site generator via chat tomorrow? If you could send me an e-mail with your Google Talk or Skype ID, we can figure out the details. I'd love for you to get involved with the coding process as well, and on that note, it'd be important for me to talk with you about style guidelines, the general code structure so far, etc. Let me know what you think!
